pandas/core/index.py

```diff
@@ -19,3 +21,9 @@
     get_objs_combined_axis,
 )
 from pandas.core.indexes.multi import _sparsify  # noqa:F401
+
+warnings.warn(
+    "pandas.core.index is deprecated and will be removed in a future version.  "
+    "Use pandas.core.indexes.api instead.",
+    FutureWarning,
```

Comment:
Let's start with a DeprecationWarning?

Reply:
this  has been around forever 
let’s just be loud about it (and it had been deprecation via documentation. forever)

Reply:
OK, opened an issue about it at fastparquet: https://github.com/dask/fastparquet/issues/470